src/pybind/mgr/dashboard/frontend/src/app/core/navigation/task-manager/task-manager.component.ts

```diff
@@ -11,38 +11,44 @@ import { TaskManagerMessageService } from '../../../shared/services/task-manager
   styleUrls: ['./task-manager.component.scss']
 })
 export class TaskManagerComponent implements OnInit {
-
-  executingTasks: Array<ExecutingTask> = [];
-  finishedTasks: Array<FinishedTask> = [];
+  executingTasks: ExecutingTask[] = [];
+  finishedTasks: FinishedTask[] = [];
 
   icon = 'fa-hourglass-o';
 
-  constructor(private summaryService: SummaryService,
-              private taskManagerMessageService: TaskManagerMessageService) {
-  }
+  constructor(
+    private summaryService: SummaryService,
+    private taskMessageManager: TaskManagerMessageService
+  ) {}
 
   ngOnInit() {
-    const icons = ['fa-hourglass-o', 'fa-hourglass-start', 'fa-hourglass-half', 'fa-hourglass-end'];
-    let iconIndex = 0;
     this.summaryService.summaryData$.subscribe((data: any) => {
-      this.executingTasks = data.executing_tasks;
-      this.finishedTasks = data.finished_tasks;
-      for (const excutingTask of this.executingTasks) {
-        excutingTask.description = this.taskManagerMessageService.getDescription(excutingTask);
-      }
-      for (const finishedTask of this.finishedTasks) {
-        finishedTask.description = this.taskManagerMessageService.getDescription(finishedTask);
-        if (finishedTask.success === false) {
-          finishedTask.errorMessage = this.taskManagerMessageService.getErrorMessage(finishedTask);
-        }
-      }
-      if (this.executingTasks.length > 0) {
-        iconIndex = (iconIndex + 1) % icons.length;
-      } else {
-        iconIndex = 0;
-      }
-      this.icon = icons[iconIndex];
+      this._handleTasks(data.executing_tasks, data.finished_tasks);
+      this._setIcon(data.executing_tasks.length);
     });
   }
 
+  _handleTasks(executingTasks: ExecutingTask[], finishedTasks: FinishedTask[]) {
+    for (const excutingTask of executingTasks) {
+      excutingTask.description = this.taskMessageManager.getRunningMessage(excutingTask);
+    }
+    for (const finishedTask of finishedTasks) {
+      finishedTask.description = this.taskMessageManager.getDescription(finishedTask);
+      if (finishedTask.success === false) {
+        finishedTask.errorMessage = this.taskMessageManager.getErrorMessage(finishedTask);
+      }
+    }
+    this.executingTasks = executingTasks;
+    this.finishedTasks = finishedTasks;
+  }
+
+  _setIcon(executingTasks: number) {
+    const iconSuffix = ['o', 'start', 'half', 'end'];
+    // TODO: Use all suffixes
```

Comment:
Shouldn't we remove this comment?

Reply:
We are currently not using all suffixes - I could remove the unused ones, but they were implemented first - so I think someone wanted to use them.

Reply:
Removed all other comments except the TODO